ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/place.hpp

```diff
@@ -17,47 +17,171 @@
+                m_inputs_tensors.push_back(input);
+            }
+
+            void addOutput(const std::weak_ptr<TensorPlacePDPD>& output, const std::string& name) {
+                m_output_names.push_back(name);
+                m_output_name_to_idx[name].push_back(m_outputs_tensors.size());
+                m_outputs_tensors.push_back(output);
+            }
+
+            const std::vector<std::weak_ptr<TensorPlacePDPD>>& getOutputTensors() const {
+                return m_outputs_tensors;
+            }
+
+            const std::vector<std::weak_ptr<TensorPlacePDPD>>& getInputTensors() const {
+                return m_inputs_tensors;
+            }
+
+            std::shared_ptr<TensorPlacePDPD> getOutputTensorByName(const std::string& name, int idx = 0) {
+                return m_outputs_tensors[m_output_name_to_idx.at(name)[idx]].lock();
+            }
+
+            std::shared_ptr<TensorPlacePDPD> getInputTensorByName(const std::string& name, int idx = 0) {
+                return m_inputs_tensors[m_input_name_to_idx.at(name)[idx]].lock();
+            }
+
+            std::string getInputNameByIdx(int idx) {
+                return m_input_names[idx];
+            }
+
+            std::string getOutputNameByIdx(int idx) {
+                return m_output_names[idx];
+            }
+
+            const void* getDesc() const { return m_op_desc; }
+
+        private:
+            // std::shared_ptr<paddle::framework::proto::OpDesc>
+            const void* m_op_desc;
+            std::map<std::reference_wrapper<const std::string>, std::vector<int>, std::less<std::string>> m_input_name_to_idx;
+            std::map<std::reference_wrapper<const std::string>, std::vector<int>, std::less<std::string>> m_output_name_to_idx;
+
+            std::vector<std::string> m_input_names;
+            std::vector<std::string> m_output_names;
+            std::vector<std::weak_ptr<TensorPlacePDPD>> m_inputs_tensors;
+            std::vector<std::weak_ptr<TensorPlacePDPD>> m_outputs_tensors;
+        };
+
+        class TensorPlacePDPD : public PDPDPlace {
+        public:
+            TensorPlacePDPD(const InputModel& input_model,
+                            const std::vector<std::string>& names,
+                            const void* var_desc)
+                    : PDPDPlace(input_model, names),
+                      m_var_desc(var_desc) {
+            }
+
+            TensorPlacePDPD(const InputModel& input_model,
+                            const void* var_desc)
+                    : TensorPlacePDPD(input_model, {}, var_desc) {
+            }
+
+            void addInput(const std::weak_ptr<OpPlacePDPD>& input) {
+                m_producing_ops.push_back(input);
+            }
+
+            void addOutput(const std::weak_ptr<OpPlacePDPD>& output) {
+                m_consuming_ops.push_back(output);
+            }
+
+            const PartialShape& getPartialShape() const { return m_pshape; }
```

Comment:
Need to have also setPartialShape and setElementType

Reply:
done